rca/wagtailapi/filters.py

```diff
@@ -19,6 +22,43 @@ def filter_queryset(self, request, queryset, view):
         return queryset
 
 
+class SubjectsFilter(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        if hasattr(queryset.model, "subjects"):
+            subject_ids = request.GET.getlist("subjects", [])
+            if subject_ids:
+                queryset = queryset.model.objects.filter(
+                    subjects__subject__in=subject_ids
+                )
+
+        return queryset
+
+
+class RelatedSchoolsFilter(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        if hasattr(queryset.model, "related_schools_and_research_pages"):
+            school_ids = request.GET.getlist("related_schools_and_research_pages", [])
+            if school_ids:
+                school_pages = (
+                    SchoolsAndResearchPage.objects.all()
+                    .live()
+                    .filter(id__in=school_ids)
+                )
+                programme_pages_with_school_pages_relationship = (
+                    ProgrammePageRelatedSchoolsAndResearchPage.objects.all()
+                )
+                relationship_objects = programme_pages_with_school_pages_relationship.filter(
+                    page__in=school_pages
+                ).values_list(
+                    "id", flat=True
+                )
+
+                queryset = queryset.model.objects.filter(
+                    related_schools_and_research_pages__id__in=relationship_objects
```

Comment:
`related_schools_and_research_pages__page_id__in=school_pages.values_list('pk', flat=True)`

Wouldn't this make the `relationship_objects` queryset redundant?

Reply:
yep you are right... I'll amend this